### PR TITLE
Add event script

### PR DESCRIPTION
**Pull request by John-Holt-Tessella**
Get all events from Database

https://github.com/ISISComputingGroup/IBEX/issues/6169

### CODE REVIEW

File: scripts/log_event_generator.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    parser.add_argument("pv_names", nargs="+", help="Each pv appearing in the data")
+
+    args = parser.parse_args()
+
+    try:
+        data_start_time = datetime.strptime(args.start_time, "%Y-%m-%dT%H:%M:%S")
+    except (ValueError, TypeError) as ex:
+        print("Can not interpret date '{}' error: {}".format(args.start_time, ex))
+        exit(1)
```
Comment (DominicOram) [inline scripts/log_event_generator.py:79]: nit: Pull this into a function

File: scripts/restore_motor_positions.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    """
+    Print a summary of collected information
+    Args:
+        db_values: values from the database
+        pv_values: values from cachannel
+        data_time: time data was asked for
+        log_file: log file to write summary to
+    """
+    print_and_log(f"PVs at time {data_time}", log_file)
+    print_and_log(f'{"name":12} {"motor":7}: {"db value":12} {"diff from":12} - {"last update":19} '
```
Comment (DominicOram) [inline scripts/restore_motor_positions.py:139]: nit: As elsewhere i feel like `historic` is better than `db`

**Review by DominicOram — CHANGES_REQUESTED** [review]
Works well, thanks. Only major issue is that the date seems to be formatted wrong. You can see it in the example in the documentation where you have a date of `2021-33-29 15:01:53`. Comments in code are minor, especially given it's just a developer tool, feel free to do as much as you like with it.

File: scripts/restore_motor_positions.py [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+
+        for pv_name, pv_value, next_change in db_values:
+            is_enabled, motor_name, value = pv_values[pv_name]
+            restore_motor_position(pv_name, pv_value, next_change, is_enabled, motor_name, value, log_file)
+
+
+if __name__ == '__main__':
+    description = "Find positions of motors in the past and restore those to current positions " \
+                  "--time 2018-01-10T09:00:00 --host ndximat " \
+                  "--prefix IN:IMAT:MOT --controller 01"
```
Comment (DominicOram) [inline scripts/restore_motor_positions.py:262]: You don't need to specify the MOT in the prefix

**Comment by DominicOram** [comment]
Also, could one of the prefix/host default to the other if only one specified. e.g. if you specify `--host IMAT` then the prefix should default to `IN:IMAT` and vis-versa. It would make the interface nicer and it's unlikely you would want to read from one instrument and set to another.

File: scripts/log_event_generator.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply scripts/log_event_generator.py:79]: good plan, done

File: scripts/restore_motor_positions.py [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+def get_motor_values(pv_names: List[str], data_time: datetime, host: str = "127.0.0.1",
+                     check_movement_for=timedelta(hours=1)) -> List[Tuple[str, ArchiverDataValue, Optional[datetime]]]:
+    """
+    Get the values for the motors from the database
+    Args:
+        pv_names: base motor names
+        data_time: time to get the data for
+        host: host to get data from
+        check_movement_for: time to check for motor movement
```
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply scripts/restore_motor_positions.py:85]: this is in the signature, which is more accurate than the comment

**Comment by John-Holt-Tessella** [comment]
> Also, could one of the prefix/host default to the other if only one specified. e.g. if you specify `--host IMAT` then the prefix should default to `IN:IMAT` and vis-versa. It would make the interface nicer and it's unlikely you would want to read from one instrument and set to another. (but still allow it by specifying both)

Yeah but this requires interrogating the inst list and in 99% of cases the user does not specify either since you can only restore motor positions on the instrument.

File: scripts/restore_motor_positions.py [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+
+    if args.controller is None:
+        controllers = range(1, 11)
+    else:
+        controllers = [int(args.controller)]
+
+    if args.prefix is None:
+        prefix = g.prefix_pv_name("")
+    else:
+        prefix = args.prefix
```
[the remark this answers is not in the document]
Comment (John-Holt-Tessella) [reply scripts/restore_motor_positions.py:291]: has to be done after set instrument which needs to be done after host is known

**Review by DominicOram — APPROVED** [review]
Great, thank you

**Pull request merged**